Taskfile.yml

```diff
@@ -0,0 +1,89 @@
+# https://taskfile.dev
+# To run on a system without task installed, `./scripts/run_task.sh` will execute it with `go run`.
+# If in the nix dev shell, `task` is available.
+
+version: '3'
+
+tasks:
+  default: ./scripts/run_task.sh --list
+
+  actionlint:
+    desc: Run the actionlint.sh
+    cmd: bash ./scripts/actionlint.sh # tests.yml
+
+  build:
+    desc: Run the build.sh
+    cmd: bash ./scripts/build.sh # tests.yml
+
+  build-antithesis-images-avalanchego:
+    desc: Build the antithesis images for the avalanchego test setup
+    cmd: bash -x ./scripts/build_antithesis_images.sh # publish_antithesis_images.yml
+
+  build-bench-precompiles:
+    desc: Build the benchmark precompiles
+    cmd: bash ./scripts/build_bench_precompiles.sh # bench.yml
+
+  build-docker-image:
+    desc: Build the docker image
+    cmd: bash ./scripts/build_docker_image.sh # publish_docker.yml
+
+  build-test:
+    desc: Run the build_test.sh
+    cmd: bash ./scripts/build_test.sh # tests.yml
+
+  coverage:
+    desc: Run the coverage.sh
+    cmd: bash ./scripts/coverage.sh # tests.yml
+
+  install-avalanchego-release:
+    desc: Install AvalancheGo release for testing
+    env:
+      BASEDIR: /tmp/e2e-test
+      AVALANCHEGO_BUILD_PATH: /tmp/e2e-test/avalanchego
```

Comment:
we should probably let CI to pass that arg

Reply:
It should definetly be possible to pass env vars from CI with `run_task` so that moving the `env` from here to `ci.yml` should work. 

I'm just not sure if we need to, but this looks to me that if we want to run these scripts in local we have no way to pass a custom path bc the task.yml just overrides it with `/tmp/e2e-test` (unless I'm mistaken and task.yml just provides a default value here).

Reply:
From reading the [order precedence](https://taskfile.dev/experiments/env-precedence/), the order in which environment variables can be determined by the user (e.g. a variable set in .bashrc vs. via the env taskfile). I am not a fan of modifying `run_task.sh` to pass environment variables because it feels very clunky, and isn't how it's configured in other repositories.

Instead, I've clarified that these are default values, and can be overridden by the user locally if they wish to run a task locally - for example `task test-e2e-precompile AVALANCHEGO_BUILD_PATH=/my/local/path DATA_DIR=/my/local/data` (the same thing `run_task.sh` does). [See this](https://taskfile.dev/usage/#variables), and of course you can always run the scripts directly.